src/main/java/io/oneko/kubernetes/impl/DeploymentManagerImpl.java

```diff
@@ -86,38 +134,40 @@ private WritableDeployment getOrCreateDeploymentForVersion(ProjectVersion<?, ?>
 				.orElseGet(() -> WritableDeployment.getDefaultDeployment(projectVersion.getId()));
 	}
 
-	private WritableProjectVersion updateDeployableWithCreatedResources(WritableProjectVersion deployable) {
+	private Optional<WritableProjectVersion> updateDeployableWithCreatedResources(WritableProjectVersion deployable) {
 		deployable.setOutdated(false);
 		if (!StringUtils.isEmpty(deployable.getDockerContentDigest())) {
-			return deployable;
+			return Optional.of(deployable);
 		}
 
 		return dockerRegistryClientFactory.getDockerRegistryClient(deployable.getProject())
 				.map(client -> {
 					final Manifest manifest = client.getManifest(deployable);
 					deployable.setDockerContentDigest(manifest.getDockerContentDigest());
 					return deployable;
-				}).orElse(null);
+				});
 	}
 
 	@Override
-	public ReadableProjectVersion stopDeployment(WritableProjectVersion version) {
-		try {
-			final WritableDeployment deployment = getOrCreateDeploymentForVersion(version);
-			HelmCommandUtils.uninstall(version);
-			deploymentRepository.deleteById(deployment.getId());
-			version.setDesiredState(NotDeployed);
-			final ReadableProject readableProject = projectRepository.add(version.getProject());
-			return readableProject.getVersions().stream()
-					.filter(projectVersion -> projectVersion.getUuid().equals(version.getId()))
-					.findFirst().orElse(null);
-		} catch (HelmRegistryException e) {
-			log.error("failed to stop deployment ({})", versionKv(version), e);
-			throw new RuntimeException(e);
-		}
+	public ReadableProjectVersion stopDeployment(final WritableProjectVersion version) {
```

Comment:
We could add a log statement here, too. Similar to the one you added in the `deploy` method.

Reply:
yes